lib/data-serializer.js

```diff
@@ -8,12 +8,29 @@ class DataSerializer {
   /**
    * Get latest split and segment data and return a script containing serialized strings.
    */
-  generateSerializedDataScript () {
-    const cachedSerializedData = this.poller.cache.serializedData ? this.poller.cache.serializedData : '{}'
+  generateSerializedDataScript (splits = []) {
+    const serializedData = splits.length > 0
+      ? this._getSomeSerializedData(splits)
+      : this._getAllSerializedData()
+
     return `<script>
-      window.__splitCachePreload = ${cachedSerializedData};
+      window.__splitCachePreload = ${serializedData};
     </script>`
   }
+
+  _getSomeSerializedData (splits) {
+    const key = splits.sort().join('.')
+    if (this.poller.cache.serializedDataSubsets[key]) {
+      return this.poller.cache.serializedDataSubsets[key]
+    }
+    const serializedData = this.poller.generateSerializedData(splits)
+    this.poller.cache.serializedDataSubsets[key] = serializedData
```

Comment:
Should we move this logic into `generateSerializedData`? I wouldn't expect `DataSerializer` to edit the poller's cache.